src/defaultCredentials.ts

```diff
@@ -0,0 +1,89 @@
+import * as fs from "fs";
+import * as path from "path";
+
+import * as api from "./api";
+import { configstore } from "./configstore";
+import * as logger from "./logger";
+
+interface RefreshTokenCredential {
+  client_id: string;
+  client_secret: string;
+  refresh_token: string;
+  type: string;
+}
+
+/**
+ * Creates Cloud application default credentials using the authorization from
```

Comment:
This class doesn't actually create anything. The methods inside it do. That said, this class doesn't hold any information that's required across all the functions - why have a class at all?

Could all the public methods in this class be top-level exported functions, and the helper (private) methods be unexported? That would eliminate the need for a class (which is preferred - these are functionally static methods, and container classes for static methods are discouraged)